### ISSUE TITLE
Semicolons are required sometimes despite NoSemicolons

### ISSUE THREAD

**Issue by Mygod** [detekt/detekt#2815]
Consider snippet where the semicolon is required:
```
val func = try {
    val input = fetchInput();
    { x -> process(input, x) }
} catch (_: Exception) {
    { x -> Unit }
}
func(x)
```

## Expected Behavior
Do not report `NoSemicolons`

## Observed Behavior
Report `NoSemicolons`

## Steps to Reproduce

See above ^

## Your Environment

* Version of detekt used: 1.8.0
* Version of Gradle used (if applicable):
* Operating System and version:
* Link to your project (if it's a public repository): https://github.com/Mygod/VPNHotspot/blob/ad218d7ec67347e18ddbd7934a0900a2da6073c8/mobile/src/main/java/be/mygod/librootkotlinx/RootServer.kt#L389

**Comment by schalkms** [comment]
This is a Ktlint rule. Does it also occur if you use plain ktlint?

**Comment by arturbosch** [comment]
@schalkms right.

It was already fixed in a newer KtLint version - https://github.com/pinterest/ktlint/issues/703.

@Mygod please try version 1.10.0-RC1.

**Comment by Mygod** [comment]
Confirming as fixed in v1.10.0. Thanks!

**Comment by Mygod** [comment]
Sorry I forgot to enable the plugin. The issue is still there in v1.10.0.

**Comment by Mygod** [comment]
It seems like it cannot handle the edge case like this: https://github.com/Mygod/VPNHotspot/blob/e333c859dfe4e675005a56049c510e5fb4ff270c/mobile/src/main/java/be/mygod/librootkotlinx/RootServer.kt#L498

**Comment by arturbosch** [comment]
> It seems like it cannot handle the edge case like this: https://github.com/Mygod/VPNHotspot/blob/e333c859dfe4e675005a56049c510e5fb4ff270c/mobile/src/main/java/be/mygod/librootkotlinx/RootServer.kt#L498

Hehe, it is a nice edge case. Might open an issue there and link it here also?
